popolo/importers/popit.py

```diff
@@ -407,8 +407,10 @@ def generate_membership_id(membership_data):
         result.end_date = membership_data.get('end_date', '')
         if 'legislative_period_id' in membership_data:
             period_data = self.events[membership_data['legislative_period_id']]
-            result.start_date = period_data.get('start_date', '')
-            result.end_date = period_data.get('end_date', '')
+            if not result.start_date:
+                result.start_date = period_data.get('start_date', '')
+            if not result.end_date:
+                result.end_date = period_data.get('end_date', '')
```

Comment:
This is still not quite right - the absence of membership start and end dates does not necessarily imply they were an MP for the entire duration of the term.